### PR TITLE
Use goreleaser to manage releases

### PR DESCRIPTION
**Pull request by zaquestion**
Will build truss for Mac (+push to homebrew tap) and linux. Will add binaries and a generated changelog to the github release

### CODE REVIEW

**Comment by zaquestion** [comment]
@lelandbatey There isn't a great way to install protoc-gen-go and other go project dependencies with brew and more so with goreleaser's formula generator. I considered opening a feature request on goreleaser to include support for the resource definition for brew formulas, but I don't think it would help much because protoc-gen-go doens't have released binaries anyways.

Beyond brew the release binaries this would generate also wouldn't have these dependencies. Thoughts on having truss check for these dependencies at runtime and offer to go get them for the user. This could be useful for go geting the truss repo itself which contains the annontations.proto users reference.

**Comment by zaquestion** [comment]
Breaking news, evidently `protoc-gen-go` made its way into brew. https://github.com/Homebrew/homebrew-core/blob/d0321a5fbe88267fa9a4ca5f7199f2294326c3fd/Formula/protoc-gen-go.rb which is great and I think means we could probably get `protoc-gen-gogo` into brew core!

**Comment by simenandre** [comment]
Hello 👋 

Are there any updates on this? I would appreciate being able to install truss with Homebrew. Anything I can help with?

**Comment by zaquestion** [comment]
@cobraz we need to get `protoc-gen-gogo` into homebrew, which IIRC was the last piece of the puzzle, short of some busy work to rewire goreleaser.

https://formulae.brew.sh/formula/protoc-gen-go (should be able to base it off of this formula) -- I've been meaning to submit a formula to homebrew core for a while.

**Comment by zaquestion** [comment]
PR'd homebrew to add gogo https://github.com/Homebrew/homebrew-core/pull/60022

**Comment by zaquestion** [comment]
Truss uses `protoc-gen-gogofaster` so we'll actually need that in brew first https://github.com/Homebrew/homebrew-core/pull/60084

**Comment by zaquestion** [comment]
Tested the resulting brew formula works and it works, which is to say truss functions. The only remaining quirk is that users still need to clone the truss repo into their gopath to leverage the annotations. Deps in general is still a problem for the built binaries... Perhaps we'll call this a step forward and document the additional steps for the different install scenarios

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    description: "Truss helps you build go-kit microservices without having to worry about writing or maintaining boilerplate code"
+
+    # Setting this will prevent goreleaser to actually try to commit the updated
+    # formula - instead, the formula file will be stored on the dist folder only,
+    # leaving the responsibility of publishing it to the user.
+    # If set to auto, the release will not be uploaded to the homebrew tap
+    # in case there is an indicator for prerelease in the tag e.g. v1.0.0-rc1
+    # Default is false.
+    skip_upload: true
```
Comment (zaquestion) [inline .goreleaser.yml]: Oh whoops, gonna need to enable that

**Comment by zaquestion** [comment]
Okay, this branch keeps getting reset to the original PR, even tho I've force pushed over it multiple times with the latest changes and I'll be honest, it's freaking me out that I don't know how that's happening. So I'm closing this and opening a new PR.

**Comment by zaquestion** [comment]
https://github.com/metaverse/truss/pull/310

**Pull request closed without merging**